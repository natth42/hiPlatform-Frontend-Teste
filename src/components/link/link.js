import React from 'react';
import PropTypes from 'prop-types';
import './link.scss';

export const Link = ({
    active,
    text,
    onClick
}) => {
    if (active) {
        return <button aria-label={`filtro atual: ${text}`} className="btn-active space-sides link-active">{text}</button>;
    }

    return (
        <button href='#'
            className="space-sides link"
            aria-label={`selecionar filtro: ${text}`}
            onClick={e => {
                e.preventDefault();
                onClick();
            }}
        >
            {text}
        </button>
    );
};

Link.propTypes = {
    text: PropTypes.string,
    active: PropTypes.bool,
    onClick: PropTypes.func
};