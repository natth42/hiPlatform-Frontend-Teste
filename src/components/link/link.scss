
.link, .link:hover {
    display: inline-block;
    color: #000;
    text-decoration: none;
    background-color: white;
    border: none;
    cursor: pointer;
}

.link:focus{
    outline: none;
}

.link::after {
    content: '';
    display: block;
    width: 0;
    height: 2px;
    background: #333;
    transition: width .3s;
}

.link:hover::after {
    width: 100%;
}

.link-active::after{
    content: '';
    display: block;
    width: 0;
    height: 2px;
    background: #1db954;
    width: 100%;
}

.btn-active{
    background-color: white;
    border: none;
    cursor: pointer;
}

.btn-active:focus{
    outline: none;
}