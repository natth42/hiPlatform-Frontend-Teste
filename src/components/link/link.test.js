import React from 'react';
import { Link } from './link';
import { shallow } from 'enzyme';

it('renders Link without crashing', () => {
   const wrapper = shallow(<Link />);
   expect(wrapper).toMatchSnapshot();
});




