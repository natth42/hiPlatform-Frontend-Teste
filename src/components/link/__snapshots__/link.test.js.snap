// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Link without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Link />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "space-sides link",
      "href": "#",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "space-sides link",
        "href": "#",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
